app.go

```diff
@@ -21,77 +21,138 @@ import (
 	"thingdust/apiserver"
 	"thingdust/apiservices"
 	"thingdust/conf"
+	"thingdust/eliona"
 	"thingdust/thingdust"
+	"time"
+
+	api "github.com/eliona-smart-building-assistant/go-eliona-api-client/v2"
 	"github.com/eliona-smart-building-assistant/go-eliona/asset"
 	"github.com/eliona-smart-building-assistant/go-utils/common"
 	"github.com/eliona-smart-building-assistant/go-utils/http"
 	"github.com/eliona-smart-building-assistant/go-utils/log"
-	"thingdust/eliona"
-	"time"
-	//api "github.com/eliona-smart-building-assistant/go-eliona-api-client/v2"
 )
 
+// For each enabled configuration, processSpaces() performs Continuous Asset Creation
+// for each project_id and space pair, and writes corresponding data to each asset.
+func processSpaces(configId int64) {
+	config, spaces, err := fetchSpacesAndSetActiveState(configId)
+	if err != nil {
+		return
+	}
+	if config.ProjIds != nil {
+		for _, projId := range *config.ProjIds {
+			for spaceName := range spaces {
+				confSpace, err := getOrCreateMappingIfNecessary(config, projId, spaceName)
+				if err != nil {
+					return
+				}
+				if confSpace != nil {
+					sendData(confSpace, spaces, spaceName)
+				}
+			}
+		}
+	}
+
+}
 
-// doAnything is the main app function which is called periodically
-func doAnything() {
-	// Fetch Spacs using thingdust api
-	request, err := http.NewRequestWithApiKey("https://demo.cust.prod.thingdust.io/api/v2/get_space_states", "X-API-KEY", "UEKNEYKACORWF9JMYBGLPOCPIBHNJUHYIAADBRQCEHQM2V7YJUSCVBFUNOWW" )
+func fetchSpacesAndSetActiveState(configId int64) (*apiserver.Configuration, thingdust.Spaces, error) {
+	config, err := conf.GetConfig(context.Background(), configId)
+	if err != nil {
+		log.Error("spaces", "Error reading configuration")
+		return nil, nil, err
+	}
+	if config.Enable == nil || !*config.Enable {
+		conf.SetConfigActiveState(context.Background(), *config, false)
+		return nil, nil, err
+	}
+	conf.SetConfigActiveState(context.Background(), *config, true)
+	log.Debug("Bug configs", "Processing space with configID: %v", config.ConfigId)
+	request, err := http.NewRequestWithApiKey(config.ApiEndpoint, "X-API-KEY", config.ApiKey)
 	if err != nil {
 		log.Error("spaces", "Error with request: %v", err)
-		return
+		return nil, nil, err
 	}
 	spaces, err1 := http.Read[thingdust.Spaces](request, time.Duration(time.Duration.Seconds(1)), true)
```

Comment:
Can you try to reuse the "err" variable. Its better than creating err1, err2, ...